server/tools/peertube-import-videos.ts

```diff
@@ -33,27 +34,64 @@ command
   .option('-U, --username <username>', 'Username')
   .option('-p, --password <token>', 'Password')
   .option('--target-url <targetUrl>', 'Video target URL')
+  .option('--tmpdir <tmpdir>', 'Working directory')
   .option('--since <since>', 'Publication date (inclusive) since which the videos can be imported (YYYY-MM-DD)', parseDate)
   .option('--until <until>', 'Publication date (inclusive) until which the videos can be imported (YYYY-MM-DD)', parseDate)
-  .option('-v, --verbose', 'Verbose mode')
+  .option('--first <first>', 'Process first n elements of returned playlist')
+  .option('--last <last>', 'Process last n elements of returned playlist')
+  .option('-v, --verbose [verbosity]', 'Verbosity, from 0 (only report errors) to 4 (for debugging), default is 2 (info)')
   .parse(process.argv)
 
 getServerCredentials(command)
   .then(({ url, username, password }) => {
     if (!program[ 'targetUrl' ]) {
-      console.error('--targetUrl field is required.')
+      log.error('--targetUrl field is required.')
 
       process.exit(-1)
     }
 
+    if (!program['tmpdir']) {
+      program[ 'tmpdir' ] = __dirname;
+    } else {
+      if (!existsSync(program[ 'tmpdir' ])) {
+        log.error('--tmpdir %s: directory does not exist or is not accessible', program[ 'tmpdir' ])
+
+        process.exit(-1)
+      }
+    }
+
+    log.setLevel('info')
+    if (program[ 'verbose' ]) {
+      switch (program[ 'verbose' ]) {
+        case '0':
+          log.setLevel('error')
+          break
+        case '1':
+          log.setLevel('warn')
+          break
+        case '2':
+          log.setLevel('info')
+          break
+        case '3':
+          log.setLevel('debug')
+          break
+        case '4':
+          log.setLevel('trace')
+          break
+        default:
+          log.setLevel('debug')
```

Comment:
Shouldn't it be `info`? (the default value as reported in the commander help)

Reply:
The default log level is set to `info` just above the `switch`, this is used when there is no `--verbose` option on the command line. When the program is called with `--verbose` (without a following parameter) the log level is set to `debug` to emulate the way it worked before this change. Using `--verbose` with the optional parameter makes it possible to select a level of verbosity.